src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts

```diff
@@ -0,0 +1,66 @@
+import { r, RecordNode, rtp, RT, fn } from "../../../r";
+import { IOrder } from "../../IOrder";
+import { pathUtils } from "@gmetrixr/gdash";
+
+/**
+ * Migrating media type for popup items
+ * Old Types: media, youtube, embed_code, sprite
+ * New Types: image, video, audio, 3d_model, pdf, youtube, markdown, embed_code, sprite
+ */
+class Migration implements IOrder {
+  execute(projectJson: unknown) {
+    const pJson = projectJson as RecordNode<RT.project>;
+    const projectF = r.record(pJson);
+
+    const allPopupElements = projectF.getAllDeepChildrenWithFilter(RT.element, el => el.props.element_type === "popup");
+
+    for (const element of allPopupElements) {
+      const elementF = r.element(element);
+      const allItems = elementF.getRecords(RT.item);
+
+      for(const item of allItems) {
+        const source = item.props.item_source as fn.Source;
```

Comment:
const itemF = r.record(item);
const source = itemF.getValueOrDefault(rtp.item.item_source) as fn.Source;
let sourceType = source.type as pathUtils.FileType;
if(!sourceType) {
  // get the file type from the file_urls
  const url = source.file_urls?.glb || source.file_urls?.gltf || source.file_urls?.o || "";
  sourceType = pathUtils.getFileType(url);
}

// switch-case